### PR TITLE
feat(internetaak-details): add partij digitale adressen fallback

### LINKED ISSUE #446: 🛠 Task: Implementeer server-side fallback-logica digitale adressen in InternetaakDetailsService
# 🛠 Task: Implementeer server-side fallback-logica in InternetaakDetailsService

---

## Intent

De medewerker ziet altijd contactgegevens als deze beschikbaar zijn — ook wanneer ze niet bij de betrokkene zelf maar bij de gekoppelde partij zijn opgeslagen (OIP-scenario).

---

## Functional Description

Implementeer fallback-verrijking in `InternetaakDetailsService`: na het laden van het klantcontact met betrokkenen, controleer voor elke betrokkene of `_expand.digitaleAdressen` leeg is. Zo ja, en `_expand.wasPartij` een UUID bevat, haal digitale adressen op via `GetPartijDigitaleAdressenAsync` en injecteer deze in de betrokkene-expand. Het client-contract (digitaleAdressen array) blijft ongewijzigd.

---

## Design Reference

N/A — geen design voor deze taak.

---

### In scope

- Fallback-logica in `InternetaakDetailsService` die partij-adressen ophalt wanneer betrokkene geen eigen adressen heeft
- Graceful degradation bij fouten (API niet bereikbaar, partij niet gevonden)
- Logging van fallback-gebruik

### Out of scope

- Wijzigingen aan `OpenKlantApiClient` methode (Task 1)
- Frontend wijzigingen
- Herkomst-indicatie tonen (expliciet niet gewenst per Feature AC)

---

### Request

Intern — geen wijziging aan het externe API contract.

### Response

Het bestaande `Internetaak`-response met nested `Klantcontact.Expand.HadBetrokkenen[].Expand.DigitaleAdressen` bevat nu ook adressen afkomstig van de partij (transparant voor de client).

### Events / Side Effects

Geen — read-only verrijking.

---

## Technical Approach

- Voeg een `EnrichBetrokkenenWithPartijDigitaleAdressenAsync` methode toe aan `InternetaakDetailsService`
- Roep deze aan na het laden van het klantcontact (na `EnrichWithZaakAsync`)
- Voor elke betrokkene in `klantcontact.Expand.HadBetrokkenen`:
  - Als `betrokkene.Expand.DigitaleAdressen` null of leeg
  - EN `betrokkene.Expand.WasPartij?.Uuid` niet null
  - → Roep `GetPartijDigitaleAdressenAsync(wasPartij.Uuid)` aan
  - → Zet resultaat in `betrokkene.Expand.DigitaleAdressen`
- Wrap in try-catch voor graceful degradation

### Aggregates & Invariants

- Aggregates: Betrokkene, Partij (read-only)
- Invariants: Eigen adressen van betrokkene worden nooit overschreven; partij-lookup wordt alleen uitgevoerd bij afwezigheid eigen adressen

---

## Impacted Areas

- Backend: `InterneTaakAfhandeling.Web.Server/Features/InterneTaakDetails/InternetaakDetailsService.cs`
- Frontend: Geen
- Database: N/A
- APIs: Open Klant API (indirect via `GetPartijDigitaleAdressenAsync`)
- External systems: Open Klant

---

## Gherkin

```gherkin
Feature: Fallback digitale adressen via partij

  Scenario: Betrokkene zonder eigen adressen met gekoppelde partij die adressen heeft
    Given een betrokkene zonder eigen digitale adressen
    And de betrokkene is gekoppeld aan een partij met een e-mailadres en telefoonnummer
    When de medewerker het contactverzoek opent
    Then ziet de medewerker het e-mailadres en telefoonnummer van de partij in het behandelscherm

  Scenario: Betrokkene met eigen digitale adressen
    Given een betrokkene met een eigen e-mailadres
    And de betrokkene is gekoppeld aan een partij met een ander e-mailadres
    When de medewerker het contactverzoek opent
    Then ziet de medewerker alleen het eigen e-mailadres van de betrokkene

  Scenario: Betrokkene zonder eigen adressen en zonder partij-identifier
    Given een betrokkene zonder eigen digitale adressen
    And de betrokkene is niet gekoppeld aan een partij
    When de medewerker het contactverzoek opent
    Then blijft de contactsectie leeg

  Scenario: Betrokkene zonder eigen adressen met partij zonder adressen
    Given een betrokkene zonder eigen digitale adressen
    And de betrokkene is gekoppeld aan een partij zonder digitale adressen
    When de medewerker het contactverzoek opent
    Then blijft de contactsectie leeg

  Scenario: Partij-lookup faalt door onbereikbare API
    Given een betrokkene zonder eigen digitale adressen
    And de betrokkene is gekoppeld aan een partij
    And de Open Klant API is niet bereikbaar voor de partij-lookup
    When de medewerker het contactverzoek opent
    Then blijft de contactsectie leeg
    And wordt de fout gelogd zonder de medewerker te hinderen
```

---

## Edge Cases & Risks

- Open Klant API onbereikbaar bij fallback → graceful degradation (log warning, geen foutmelding naar gebruiker)
- Betrokkene heeft `WasPartij` met UUID maar partij bestaat niet → behandel als lege adressen
- Meerdere betrokkenen bij hetzelfde klantcontact → fallback per betrokkene afzonderlijk uitvoeren
- Performance: extra API-call alleen wanneer nodig (lazy fallback)

---

## Observability

- Logs: `LogInformation` wanneer fallback wordt uitgevoerd, `LogWarning` bij falen van partij-lookup
- Metrics: Geen nieuwe metrics
- Alerts: Geen

---

## Dependencies

Depends on Task 1 (#445): `GetPartijDigitaleAdressenAsync` methode moet beschikbaar zijn.

---

## Rollout

- Feature flag: N/A
- Backward compatibility: Volledig backward-compatible — bestaand gedrag ongewijzigd wanneer betrokkene eigen adressen heeft
- Data migration: N/A

---

## Definition of Done

- [ ] Contracts match implementation
- [ ] All Gherkin scenarios covered by tests (Phase 2 E2E)
- [ ] Edge cases handled
- [ ] Observability in place
- [ ] Code reviewed
- [ ] Linting passes (`npm run lint:ci`)
- [ ] Formatting passes (`npm run format:ci`)

---

## Test Mapping

| Gherkin Scenario | Test file | Status |
| ---------------- | --------- | ------ |
| Betrokkene zonder eigen adressen met gekoppelde partij die adressen heeft | Phase 2 E2E | Pending |
| Betrokkene met eigen digitale adressen | Phase 2 E2E | Pending |
| Betrokkene zonder eigen adressen en zonder partij-identifier | Phase 2 E2E | Pending |
| Betrokkene zonder eigen adressen met partij zonder adressen | Phase 2 E2E | Pending |
| Partij-lookup faalt door onbereikbare API | Phase 2 E2E | Pending |

### PR DESCRIPTION
**Pull request by nijmra**
## Summary

De medewerker ziet altijd contactgegevens als deze beschikbaar zijn — ook wanneer ze bij de gekoppelde partij zijn opgeslagen (OIP-scenario). Deze task implementeert de server-side fallback in `InternetaakDetailsService`: wanneer een betrokkene geen eigen digitale adressen heeft maar wel een partij-identifier, worden de adressen van de partij opgehaald en transparant ingeject in de betrokkene-expand.

## Changes

- **Unit test project**: Nieuw `InterneTaakAfhandeling.Web.Server.Test` project met 5 unit tests die alle Gherkin scenarios afdekken
- **Fallback verrijking**: `EnrichBetrokkenenWithPartijDigitaleAdressenAsync` methode in `InternetaakDetailsService` — itereert over betrokkenen, detecteert lege adressen + aanwezige partij-identifier, en verrijkt via `GetPartijDigitaleAdressenAsync`
- **Graceful degradation**: try-catch per betrokkene — bij API-falen blijft de contactsectie leeg zonder foutmelding naar de gebruiker

## Test plan

- [x] Betrokkene zonder eigen adressen met gekoppelde partij die adressen heeft → partij-adressen getoond
- [x] Betrokkene met eigen digitale adressen → alleen eigen adressen, partij niet geraadpleegd
- [x] Betrokkene zonder eigen adressen en zonder partij-identifier → leeg, geen lookup
- [x] Betrokkene zonder eigen adressen met partij zonder adressen → leeg
- [x] Partij-lookup faalt door onbereikbare API → leeg, fout gelogd

## Related

Closes #446

### CODE REVIEW

**Review by nijmra — COMMENTED** [review]
## PR Review — #449: feat(internetaak-details): add partij digitale adressen fallback

**Linked task:** #446 | **Spec coverage:** 5/5 scenarios covered

| Dimension | Result | Notes |
|-----------|--------|-------|
| Spec adherence | ✅ PASS | All 5 Gherkin scenarios have matching unit tests |
| Contract compliance | ✅ PASS | Response shape unchanged; partij adressen transparently injected |
| Impacted Areas | ✅ PASS | Changes within declared scope |
| Code quality | ✅ PASS | Follows vertical-slice pattern, structured logging, <40 lines |
| Test coverage | ⚠️ WARN | (1) No test for `null` digitaleAdressen (only empty list tested); (2) failure test doesn't assert LogWarning was called |
| Error handling | ⚠️ WARN | `catch(Exception)` swallows `OperationCanceledException` |

### Verdict: APPROVE WITH COMMENTS

---

### Suggested (non-blocking):

1. **[Error handling]** Re-throw `OperationCanceledException` before the general catch — swallowing it masks shutdown/cancellation signals:
   ```csharp
   catch (OperationCanceledException) { throw; }
   catch (Exception ex) { _logger.LogWarning(...); }
   ```

2. **[Test]** Add a test where `digitaleAdressen` is `null` (not just empty list) — the code handles both paths but only `[]` is tested. Prevents regression if the null-check is accidentally removed.

3. **[Test]** Assert `LogWarning` is called in `Get_HandlesPartijLookupFailure_Gracefully` — the Gherkin explicitly requires "wordt de fout gelogd zonder de medewerker te hinderen".

---

Clean, well-structured PR. Implementation matches the spec precisely and follows established patterns.

File: InterneTaakAfhandeling.Web.Server/Features/InterneTaakDetails/InternetaakDetailsService.cs [hunk 1]
```diff
@@ -86,5 +88,37 @@ private async Task EnrichWithZaakAsync(Internetaak? interntaak)
...
+                    var partijAdressen = await _openKlantApiClient.GetPartijDigitaleAdressenAsync(partijUuid);
+
+                    betrokkene.Expand ??= new BetrokkeneExpand();
+                    betrokkene.Expand.DigitaleAdressen = partijAdressen;
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogWarning(ex, "Fallback partij-lookup gefaald voor partij {PartijUuid}: {Message}",
+                        partijUuid, ex.Message);
+                }
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.Web.Server/Features/InterneTaakDetails/InternetaakDetailsService.cs:124]: Fixed — added a catch (OperationCanceledException) { throw; } before the general catch. Also added a dedicated test to verify it propagates correctly.

File: InterneTaakAfhandeling.Web.Server.Test/Features/InterneTaakDetails/InternetaakDetailsServiceFallbackTests.cs [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+
+        // Act
+        var result = await service.Get(new InterneTaakQuery { Nummer = "123" });
+
+        // Assert
+        var betrokkene = result!.AanleidinggevendKlantcontact.Expand!.HadBetrokkenen!.First();
+        Assert.Empty(betrokkene.Expand!.DigitaleAdressen!);
+    }
+
+    [Fact]
```
[the remark this answers is not in the document]
Comment (nijmra) [reply InterneTaakAfhandeling.Web.Server.Test/Features/InterneTaakDetails/InternetaakDetailsServiceFallbackTests.cs:147]: Fixed — added a test for the null case (Get_EnrichesBetrokkeneWithPartijAdressen_WhenDigitaleAdressenIsNull). Also updated the helper method signature to accept List<DigitaleAdres>? to support this scenario.
